pkg/client/cache/reflector.go

```diff
@@ -285,7 +287,11 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 			return nil
 		}
 		if err := r.watchHandler(w, &resourceVersion, resyncCh, stopCh); err != nil {
-			if err != errorResyncRequested && err != errorStopRequested {
+			if err == errorResyncRequested {
+				glog.V(1).Infof("%s: watch of %v ending because handler requested resync", r.name, r.expectedType)
+			} else if err == errorStopRequested {
+				glog.V(1).Infof("%s: watch of %v ending because handler requested stop", r.name, r.expectedType)
+			} else {
```

Comment:
isn't the old message sufficient since it logs the error?
Both these are client controlled, i.e if we hit ResyncRequested it means the controller asked for an explicity resync (or said resync every 10s), and StopRequested means the controller asked to shutdown the watch. Both pretty unlikely I think, the clients shouldn't be the ones requesting resync, I think the server controls this.